cot/src/db/query.rs

```diff
@@ -137,6 +191,25 @@ impl<T: Model> Query<T> {
         db.get(self).await
     }
 
+    /// Execute the query and return the number of results.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if the query fails.
+    pub async fn count(&self, db: &Database) -> db::Result<usize> {
```

Comment:
This part I'm not 100% if I'm doing it correctly.

Reply:
Looks fine to me, just have a look at my comment below.